sdk/core/core/test/inc/az_test_span.h

```diff
@@ -50,22 +49,23 @@ AZ_NODISCARD AZ_INLINE az_span az_span_for_test_init(uint8_t* ptr, int32_t lengt
  * @param[in] result_span Span that has the result of the test run.
  * @param[in] buffer_expected Buffer that contains expected results of the test and that result_span
  * will match on success.
- * @param[in] length_expected The expected length of result_span.
- * @param[in] capacity_expected The expected capacity of result_span.
+ * @param[in] size_expected The expected size of result_span.
  */
 AZ_INLINE void az_span_for_test_verify(
     az_span result_span,
     const void* const buffer_expected,
     int32_t length_expected,
-    int32_t capacity_expected)
+    az_span original_buffer,
+    int32_t size_expected)
```

Comment:
@jspaith - please take a look - I tried to keep the semantics of this helper as much as possible, but maybe we don't need the overflow checking now that span doesn't have length/capacity.

Reply:
I find it a bit confusing: there's `length_expected` without documentation, size_expected, original buffer.
That said, I think this may be going away with the char*  conversion for output strings instead of az_span.